src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -75,6 +76,11 @@ def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         logger.debug("AdalAuthentication.get_token invoked by Track 2 SDK with scopes=%s", scopes)
 
         _, token, full_token, _ = self._get_token(_try_scopes_to_resource(scopes))
+
+        if 'expiresOn' in full_token:
+            return AccessToken(token, int(time.mktime(
+                datetime.datetime.strptime(full_token['expiresOn'], '%Y-%m-%d %H:%M:%S.%f').timetuple())))
```

Comment:
[`datetime.timestamp()`](https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp) is an easier way to acquire epoch time.

Reply:
Good idea, updated.